OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -124,7 +124,7 @@ protected virtual bool CanAttack(Actor self, Target target)
 
 		public virtual void DoAttack(Actor self, Target target, IEnumerable<Armament> armaments = null)
 		{
-			if (armaments == null && !CanAttack(self, target))
+			if (armaments == null || !CanAttack(self, target))
```

Comment:
Ah, right. This change was actually bogus, because `armaments` is an override, and we should actually fall back to `Armaments` with capital 'A' if the override is null.

Reply:
This whole method looks bogus - why should we check `CanAttack` in the default case, but not when given a custom armament list? This check is 50% of the method!

Reply:
...Right, this should simply check CanAttack regardless of whether the armaments override is null. I hope that will be the last update...

Reply:
It seems that there is no change to that line at all in the commit now? Also, I tested ingame with
 `if (!CanAttack(self, target))` and that fixes the bug.

Reply:
Yeah, I hadn't pushed that update yet. Done now.